pkgs/dkg/drand.go

```diff
@@ -487,14 +500,46 @@ func GetNetworkByFork(fork [4]byte) eth2_key_manager_core.Network {
 	}
 }
 
+// broadcastError propagates the error at operator back to initiator
 func (o *LocalOwner) broadcastError(err error) {
 	errMsgEnc, _ := json.Marshal(err.Error())
 	errMsg := &wire.Transport{
 		Type:       wire.ErrorMessageType,
-		Identifier: o.data.ReqID,
+		Identifier: o.Data.ReqID,
 		Data:       errMsgEnc,
 	}
-
 	o.Broadcast(errMsg)
-	close(o.done)
+	close(o.Done)
+}
+
+// checkOperators checks that operator received all participating parties DKG public keys
+func (o *LocalOwner) checkOperators() bool {
+	for _, op := range o.Data.Init.Operators {
+		if o.Exchanges[op.ID] == nil {
+			return false
+		}
+	}
+	return true
+}
+
+func (o *LocalOwner) storeSecretShareToFile(outputPath string, index int, secretKeyBLS *bls.SecretKey, validatorPubKey *bls.PublicKey) error {
+	type shareStorage struct {
+		Index  int    `json:"index"`
+		Secret string `json:"secret"`
+	}
+	// Encrypt before storing to file
+	rawKey := secretKeyBLS.SerializeToHexStr()
+	encryptedSecretShare, err := o.EncryptFunc([]byte(rawKey))
+	if err != nil {
+		return fmt.Errorf("cant encrypt private share")
+	}
+	data := shareStorage{
+		Index:  index,
+		Secret: hex.EncodeToString(encryptedSecretShare),
+	}
+	err = utils.WriteJSON(outputPath+"secret_share_"+fmt.Sprintf("%d", data.Index)+"_"+validatorPubKey.SerializeToHexStr(), &data)
```

Comment:
I can't see where this is loaded? also lets save move `secret_share` to a constant and create the whole string using `fmt.Sprintf`